pandas/tests/reshape/test_from_dummies.py

```diff
@@ -0,0 +1,350 @@
+    )
+    expected = DataFrame({"": ["a", "b", "c", "a"]})
+    result = from_dummies(dummies)
+    tm.assert_frame_equal(result, expected)
+
+
+def test_from_dummies_no_prefix_int_cats_basic():
+    dummies = DataFrame(
+        {1: [1, 0, 0, 0], 25: [0, 1, 0, 0], 2: [0, 0, 1, 0], 5: [0, 0, 0, 1]}
+    )
+    expected = DataFrame({"": [1, 25, 2, 5]}, dtype="object")
+    result = from_dummies(dummies)
+    tm.assert_frame_equal(result, expected)
+
+
+def test_from_dummies_no_prefix_float_cats_basic():
+    dummies = DataFrame(
+        {1.0: [1, 0, 0, 0], 25.0: [0, 1, 0, 0], 2.5: [0, 0, 1, 0], 5.84: [0, 0, 0, 1]}
+    )
+    expected = DataFrame({"": [1.0, 25.0, 2.5, 5.84]}, dtype="object")
+    result = from_dummies(dummies)
+    tm.assert_frame_equal(result, expected)
+
+
+def test_from_dummies_no_prefix_mixed_cats_basic():
+    dummies = DataFrame(
+        {
+            1.23: [1, 0, 0, 0, 0],
+            "c": [0, 1, 0, 0, 0],
+            2: [0, 0, 1, 0, 0],
+            False: [0, 0, 0, 1, 0],
+            None: [0, 0, 0, 0, 1],
+        }
+    )
+    expected = DataFrame({"": [1.23, "c", 2, False, None]}, dtype="object")
+    result = from_dummies(dummies)
+    tm.assert_frame_equal(result, expected)
+
+
+def test_from_dummies_no_prefix_string_cats_contains_get_dummies_NaN_column():
+    dummies = DataFrame({"a": [1, 0, 0], "b": [0, 1, 0], "NaN": [0, 0, 1]})
+    expected = DataFrame({"": ["a", "b", "NaN"]})
+    result = from_dummies(dummies)
+    tm.assert_frame_equal(result, expected)
+
+
+def test_from_dummies_no_prefix_contains_unassigned():
+    dummies = DataFrame({"a": [1, 0, 0], "b": [0, 1, 0]})
+    with pytest.raises(
+        ValueError,
+        match=(
+            r"Dummy DataFrame contains unassigned value\(s\); "
+            r"First instance in row: 2"
+        ),
+    ):
+        from_dummies(dummies)
+
+
+def test_from_dummies_no_prefix_string_cats_base_category():
+    dummies = DataFrame({"a": [1, 0, 0], "b": [0, 1, 0]})
+    expected = DataFrame({"": ["a", "b", "c"]})
+    result = from_dummies(dummies, base_category="c")
+    tm.assert_frame_equal(result, expected)
+
+
+def test_from_dummies_no_prefix_wrong_base_category_type():
+    dummies = DataFrame({"a": [1, 0, 1], "b": [0, 1, 1]})
+    with pytest.raises(
+        TypeError,
+        match=(
+            r"Expected 'base_category' to be of type 'None', 'Hashable', or 'dict'; "
+            r"Received 'base_category' of type: list"
+        ),
+    ):
+        from_dummies(dummies, base_category=["c", "d"])
+
+
+def test_from_dummies_no_prefix_multi_assignment():
+    dummies = DataFrame({"a": [1, 0, 1], "b": [0, 1, 1]})
+    with pytest.raises(
+        ValueError,
+        match=(
+            r"Dummy DataFrame contains multi-assignment\(s\); "
+            r"First instance in row: 2"
+        ),
+    ):
+        from_dummies(dummies)
+
+
+def test_from_dummies_no_prefix_contains_nan():
```

Comment:
can you organize the tests that raise and put them together 

also name them like

test_error_no_prefix_contains_nan or similar

(dont' need to use from_dummies) on every test (or really any of them) as this is the from_dummies file